main.cpp

```diff
@@ -58,46 +60,72 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    int value;
+    std::string name;
+
+    T(int v, const char* charA) :    
+    value(v),
+    name(charA)
+    {
+        
+    }
 };
 
-struct <#structName1#>                                //4
+struct Comparison
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b)
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if(a != nullptr && b != nullptr)
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float num1 { 0 }, num2 { 0 };
+    float updateNums(float* updated)      //12
     {
+        if(updated != nullptr)
+        {
+            std::cout << "U's num1 value: " << num1 << std::endl;
+            num1 = *updated;
+            std::cout << "U's num1 updated value: " << num1 << std::endl;
         
+            while( std::abs(num2 - num1) > 0.001f )
+            {
+                num2 += 1.f;
+                //I see that the other students have used this expression (that->num1 < that->num2) ? -0.1f : 0.1f;
+                //but I don't understand it or how to even come up with it so I did not use it.
+            }
+            std::cout << "U's num2 updated value: " << num2 << std::endl;
+        }   
+        return num2 * num1;
     }
 };
 
-struct <#structname2#>
+struct UpdateThat
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float updateNums(U* that, float* updated )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if(that != nullptr)
```

Comment:
What do we always do before using pointers?
- [ ] revise

Reply:
Oh shoot!!! you said it about a million times and one still slipped by me